dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/BitstreamAccessStatusLinkRepository.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+
+package org.dspace.app.rest.repository;
+
+import java.sql.SQLException;
+import java.util.Date;
+import java.util.UUID;
+
+import jakarta.annotation.Nullable;
+import jakarta.servlet.http.HttpServletRequest;
+import org.dspace.access.status.DefaultAccessStatusHelper;
+import org.dspace.access.status.service.AccessStatusService;
+import org.dspace.app.rest.model.AccessStatusRest;
+import org.dspace.app.rest.model.BitstreamRest;
+import org.dspace.app.rest.projection.Projection;
+import org.dspace.content.Bitstream;
+import org.dspace.content.service.BitstreamService;
+import org.dspace.core.Context;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Component;
+
+/**
+ * Link repository for calculating the access status of a Bitstream,
+ * including the embargo date
+ */
+@Component(BitstreamRest.CATEGORY + "." + BitstreamRest.PLURAL_NAME + "." + BitstreamRest.ACCESS_STATUS)
+public class BitstreamAccessStatusLinkRepository extends AbstractDSpaceRestRepository
+    implements LinkRestRepository {
+
+    @Autowired
+    BitstreamService bitstreamService;
+
+    @Autowired
+    AccessStatusService accessStatusService;
+
+    @PreAuthorize("hasPermission(#bitstreamId, 'BITSTREAM', 'METADATA_READ')")
+    public AccessStatusRest getAccessStatus(@Nullable HttpServletRequest request,
+                                            UUID bitstreamId,
+                                            @Nullable Pageable optionalPageable,
+                                            Projection projection) {
+        try {
+            Context context = obtainContext();
+            Bitstream bitstream = bitstreamService.find(context, bitstreamId);
+            if (bitstream == null) {
+                throw new ResourceNotFoundException("No such bitstream: " + bitstreamId);
+            }
+            AccessStatusRest accessStatusRest = new AccessStatusRest();
+            Date availabilityDate = accessStatusService.getAvailabilityDateFromBitstream(context, bitstream);
+            String status = accessStatusService.getAccessStatusFromAvailabilityDate(availabilityDate);
```

Comment:
The access status only ever says "embargo" for users that don't have read access, unlike the access status for Items.

I understand the rationale behind this, but IMO the behaviour should remain in sync. 
Maybe we could include a configuration property to toggle it?

Reply:
I'm not sure about what you mean here.

Yes, there's a difference on how we calculate the status for items VS bitstreams.

That difference was to preserve the existing behavior on items : calculate the access status based on the primary or the first bitstream and only for the anonymous group.

And also to add a [feature](https://github.com/DSpace/dspace-angular/issues/2413#issuecomment-1747647298) required by the community on bitstreams : show the embargo for the current user (not only limited on the anonymous group).

I agree it can be a bit confusing. How should we handle this? By adding some configurations as you suggested or by unifying the behaviors?